lib/ae_mdw/db/async_store.ex

```diff
@@ -19,17 +19,17 @@ defmodule AeMdw.Db.AsyncStore do
   @typep key() :: Database.key()
   @typep record() :: Database.record()
   @typep table() :: Database.table()
-  @opaque t() :: %__MODULE__{tid: :single_async_store}
+  @type t() :: %__MODULE__{tid: :single_async_store}
```

Comment:
Why do we need to expose the internal structure? I'd rather not

Reply:
Otherwise dialyzer didn't allow using pattern match on the struct itself (not only on the properties, probably because the struct is in a property) like in here: https://github.com/aeternity/ae_mdw/pull/1732/files#diff-50003680f6579cccc558e6fd7799dad10972701a70a1a307a18f977e94634e07R32 and several other places. I prefer having explicit matches, because this will make it fail sooner and more obvious, if something else gets passed, but I can change it.

Reply:
I'd rather keep structures encapsulated in their own modules than exposing them for runtime checks. We try to avoid runtime errors and prioritize encapsulation and dialyzer (static) checks. If anyone tries sending anything other than an AsyncStore is going to get caught even before executing the app